src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let previous = initialValue;
+    let startIndex = 0;
+
+    if (arguments.length < 2) {
+      previous = this[0];
+      startIndex = 1;
+    }
```

Comment:
You could use ternary operator and rewrite this code to only 2 lines.

Reply:
ty, havent mentioned this yet, fixed it! ;)